packages/sqrl-engine/src/function/Instance.ts

```diff
@@ -396,10 +396,10 @@ export class SqrlInstance {
     }
 
     let isAsync;
-    if (fn instanceof AsyncFunction) {
+    if (hasAsyncFunction && fn instanceof AsyncFunction) {
       invariant(
-        props.async || !props.hasOwnProperty("async"),
-        "Expected async to be true for async function"
+        props.async,
+        "Async function was missing `async` property: " + name
       );
       fn = bluebird.method(fn);
```

Comment:
why do we need this?

Reply:
I guess we don't, it's just kind of bad when a function SQRL thinks is `async` is in fact not. I'd be okay with removing this check.